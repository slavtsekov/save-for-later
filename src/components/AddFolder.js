import React from 'react';
import { connect } from 'react-redux';
import { addFolder } from '../actions/folders';

export class AddFolder extends React.Component {
    state = {
        error: undefined
    };

    addFolder = (e) => {
        e.preventDefault();

        const folderName = e.target.elements.folder.value.trim();
        if (!folderName) {
            this.setState(() => ({
                error: 'Please provide folder name.'
            }));
        } else {
            this.setState(() => ({
                error: undefined
            }));

            this.props.addFolder(folderName);
            e.target.elements.folder.value = '';
        }
    };

    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form
                    onSubmit={this.addFolder}
                >
                    <input type="text" name="folder" />
                    <button type="submit">Add Folder</button>
                </form>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    addFolder: (folder) => { dispatch(addFolder(folder)); }
});

export default connect(undefined, mapDispatchToProps)(AddFolder);
