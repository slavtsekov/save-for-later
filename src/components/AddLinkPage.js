import React from 'react';

import LinkForm from './LinkForm';

class AddLinkPage extends React.Component {
    onSubmit = (data) => {
        console.log(data);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <h1>Add Link</h1>
                <LinkForm onSubmit={this.onSubmit} />
            </div>
        );
    }
}

export { AddLinkPage as default };
