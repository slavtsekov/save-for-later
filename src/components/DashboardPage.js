import React from 'react';

const DashboardPage = () => (
    <div>
        Dashboard page content.
    </div>
);

export { DashboardPage as default };
