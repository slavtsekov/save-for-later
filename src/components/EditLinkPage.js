import React from 'react';

const EditLinkPage = () => (
    <div>
        Edit Link page content.
    </div>
);

export { EditLinkPage as default };
