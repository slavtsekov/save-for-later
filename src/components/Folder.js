import React from 'react';

const Folder = (props) => (
    <div>
        <p>{props.text}</p>
        <button
            type="button"
            onClick={(ะต) => {
                props.onDelete(props.text);
            }}
        >
            Remove
        </button>
    </div>
);

export { Folder as default };
