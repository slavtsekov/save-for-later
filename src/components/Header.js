import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = (props) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard">
                    <h1>Save For Later</h1>
                </Link>
                <button
                    className="button button--link"
                    type="button"
                    onClick={props.startLogout}
                >
                    Logout
                </button>
            </div>
            <nav>
                <Link to="/dashboard">
                    <h2>Home</h2>
                </Link>
                <Link to="/folders">
                    <h2>Folders</h2>
                </Link>
            </nav>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
