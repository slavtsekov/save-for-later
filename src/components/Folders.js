import React from 'react';

import Folder from './Folder';

class Folders extends React.Component {
    state = {
        folders: ['Folder 1', 'Folder 2']
    };

    handleDeleteFolder = (folderName) => {
        this.setState((prevState) => {
            const folders = prevState.folders.filter(folder => folder !== folderName);
            return { folders };
        });
    };

    render() {
        return (
            <div>
                <div>
                    <h3>Your Folders</h3>
                </div>
                {this.state.folders.length === 0 && <p>You have no folders at the moment.</p>}
                {
                    this.state.folders.map((folder, index) => (
                        <Folder
                            key={folder}
                            text={folder}
                            onDelete={this.handleDeleteFolder}
                        />
                    ))
                }
            </div>
        );
    }
}

export { Folders as default };
