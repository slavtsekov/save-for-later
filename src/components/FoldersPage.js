import React from 'react';
import Folders from './Folders';
import AddFolder from './AddFolder';

const FoldersPage = () => (
    <div>
        <Folders />
        <AddFolder />
    </div>
);

export { FoldersPage as default };
