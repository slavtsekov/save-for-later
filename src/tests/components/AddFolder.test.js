import React from 'react';
import { shallow } from 'enzyme';
import { AddFolder } from '../../components/AddFolder';

test('should render AddFolder correctly', () => {
    const wrapper = shallow(<AddFolder />);
    expect(wrapper).toMatchSnapshot();
});

test('should dispatch to store if folder value is provided', () => {
    const value = 'Books';
    const addFolderSpy = jest.fn();
    const wrapper = shallow(<AddFolder addFolder={addFolderSpy} />);

    wrapper.find('form').simulate('submit', {
        preventDefault: () => {},
        target: {
            elements: {
                folder: {
                    value
                }
            }
        }
    });

    expect(addFolderSpy).toHaveBeenLastCalledWith(value);
    expect(wrapper.state('error')).toBeUndefined();
});

test('should not dispatch to store if folder value is not provided', () => {
    const value = '  ';
    const addFolderSpy = jest.fn();
    const wrapper = shallow(<AddFolder addFolder={addFolderSpy} />);

    wrapper.find('form').simulate('submit', {
        preventDefault: () => {},
        target: {
            elements: {
                folder: {
                    value
                }
            }
        }
    });

    expect(addFolderSpy).not.toHaveBeenCalled();
    expect(wrapper.state('error')).toBe('Please provide folder name.');
});
