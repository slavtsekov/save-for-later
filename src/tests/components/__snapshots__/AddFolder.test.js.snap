// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddFolder correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="folder"
      type="text"
    />
    <button
      type="submit"
    >
      Add Folder
    </button>
  </form>
</div>
`;
